josephlatina/DungeonCrawler
# Health and Damage systems

- Added Health scripts for both player and enemy (PlayerHealth and EnemyHealth)
- The player takes damage from colliding with an enemy
- Created a rudimentary health bar that updates as the player gains/loses health
    - Each 0.5 health = 1/2 of a heart
- The player can do damage to enemies with both melee and ranged weapons based on the weapon's damage
    - Enemies should be knocked back slightly on hit
- Player can heal lost health by using a health potion
- Created a rudimentary death screen that appears when the player loses all their health
- Added sounds for getting hurt and healing.

To Test:
- Move the player character until it collides with an enemy. The health bar should decrease on contact.
- Pick up a ranged and a melee weapon and attack an enemy. The enemy should lose health (confirm by looking at the Current Health Points in the Enemy Health script in the inspector). The enemy should also be knocked back slightly.
- After losing some health, pick up a health potion and press F. The health bar should increase.
- Collide with an enemy until the health bar decreases completely. Game Over screen should appear.
